pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/StageNodeVisitor.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner.stage;
+
+public interface StageNodeVisitor<T, C> {
```

Comment:
1. javadoc, esp. explain the generic meaning, `T` as return type, and `C` as context
2. be sure to make the definition "context" clear. 
  - it is not clear what context it is holding, 
  - is the visit method allow to change the context 
  - in what way visit method should be changing the context